### DISCUSSION TITLE
include sessions and cache table in db backup - truncate instead of drop

### DISCUSSION THREAD

**Discussion by matthewstick** [craftcms/cloud#34] [Feature Requests & Enhancements]
We chatted this once in an email, but wanted to log it here.

So the Database backup in cloud is great, except it doesn't include the session and cache tables.  It was explained to me something like "this is because you don't need that info locally" 

Agreed we don't need the data in those tables, but we still do need the structure of the tables locally.  So right now every time we restore a db, we get errors and we have to re-run this locally:
php craft setup/php-session-table
php craft setup/db-cache-table

Except, those commands also gives errors unless we go into the app.php and remove the sessions and db php config code, then re-add it.

So it's kind of a PITA.  It's easy-ish for me, but it's a common enough task that keeps railing me.

And for other more junior people on my team, they lose sometimes hours suffering trying to figure this out, then have to call me, then I have to explain it, then they forget next time because they are still junior and not sr ;) 

If the tables were truncated instead of dropped we wouldn't have to do this.

Thanks!

**Comment by timkelty** [comment]
Thanks for the reminder, @matthewstick.
FWIW, this is specifically a problem for you because you are using DB cache/sessions in all envs, even local.

One way around it would be to just drop those customized components entirely, unless there is a specific reason you need them for local dev. Usually using the default (filesystem) for cache/sessions works fine locally, and will be faster.

**Reply by matthewstick** [reply]
You mean you do this automatically in craft cloud? If so I'm deleting this code locally!

**Reply by timkelty** [reply]
Hah, yes! You don't need to have this code at all – we take care of it for you.
